tests/e2e/e2e_test.go

```diff
@@ -17,46 +18,587 @@
+	return fmt.Sprintf("Node %q matched unexpectedly", m.node.Name)
+}
+
+// TODO remove nolint when golangci-lint is able to cope with generics
+//
+//nolint:unused
+func (m *nodeListPropertyRegexpMatcher[T]) message() string {
+	msg := fmt.Sprintf("Node %q %s did not match:", m.node.Name, m.propertyName)
+	if len(m.missing) > 0 {
+		msg += fmt.Sprintf("\n  missing:\n    %s", strings.Join(m.missing, "\n    "))
+	}
+	if len(m.invalidValue) > 0 {
+		msg += fmt.Sprintf("\n  invalid value:\n    %s", strings.Join(m.invalidValue, "\n    "))
+	}
+	return msg
+}
+
+// jobIsCompleted checks if a job is completed
+//
+//nolint:unused
+func jobIsCompleted(ctx context.Context, cli clientset.Interface, namespace, podName string) bool {
+	pod, err := cli.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+	if err != nil {
+		return false
+	}
+	// Check if the pod's phase is Succeeded.
+	if pod.Status.Phase == "Succeeded" {
+		return true
+	}
+	return false
+}
 
-	ginkgo.RunSpecs(t, "nvidia k8s-device-plugin e2e suite", suiteConfig, reporterConfig)
+// randomSuffix provides a random sequence to append to pods,services,rcs.
+//
+//nolint:unused
+func randomSuffix() string {
+	return strconv.Itoa(rand.Intn(10000))
+}
+
+// getNonControlPlaneNodes gets the nodes that are not tainted for exclusive control-plane usage
+//
+//nolint:unused
+func getNonControlPlaneNodes(ctx context.Context, cli clientset.Interface) ([]corev1.Node, error) {
+	nodeList, err := cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if len(nodeList.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found in the cluster")
+	}
+
+	controlPlaneTaint := corev1.Taint{
+		Effect: corev1.TaintEffectNoSchedule,
+		Key:    "node-role.kubernetes.io/control-plane",
+	}
+	out := []corev1.Node{}
+	for _, node := range nodeList.Items {
+		if !taintExists(node.Spec.Taints, &controlPlaneTaint) {
+			out = append(out, node)
+		}
+	}
+
+	if len(out) == 0 {
+		return nil, fmt.Errorf("no non-control-plane nodes found in the cluster")
+	}
+	return out, nil
+}
+
+// taintExists checks if the given taint exists in list of taints. Returns true if exists false otherwise.
+//
+//nolint:unused
+func taintExists(taints []corev1.Taint, taintToFind *corev1.Taint) bool {
+	for _, taint := range taints {
+		if taint.MatchTaint(taintToFind) {
+			return true
+		}
+	}
+	return false
+}
+
+// getNode returns the node object from the list of nodes
+//
+//nolint:unused
+func getNode(nodes []corev1.Node, nodeName string) corev1.Node {
+	for _, node := range nodes {
+		if node.Name == nodeName {
+			return node
+		}
+	}
+	return corev1.Node{}
+}
+
+// CreateOrUpdateJobsFromFile creates or updates jobs from a file
+func CreateOrUpdateJobsFromFile(ctx context.Context, cli clientset.Interface, filename, namespace string) ([]string, error) {
+	jobs, err := newJobFromfile(filepath.Join(packagePath, "data", filename))
```

Comment:
My one question would be why don't we provide the full relative path to the file instead of ASSUMING `data`? Searching for `data/job-1.yaml` when we do decide to move these files (or add new ones) will be simpler than searching for each path component separately.